src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1268,4 +1289,99 @@ ss::future<bool> tx_gateway_frontend::try_create_tx_topic() {
       });
 }
 
+void tx_gateway_frontend::expire_old_txs() {
+    (void)ss::with_gate(_gate, [this] {
+        auto shard = _shard_table.local().shard_for(model::tx_manager_ntp);
+
+        if (shard == std::nullopt) {
+            rearm_expire_timer();
+            return ss::now();
+        }
+
+        return container()
+          .invoke_on(
```

Comment:
can you comment on the safety of shutting down the service wrt to running this cross core operation?

when the service is stopped, there is not guarantee that the gate will close (ensuring that the cross-core operation is complete) before the core _running_ the cross core operation is shutdown. it would be safe if the originating core and the target core are the same, but otherwise it might lead to a segfault where `self` on the destination core races with service stop.

Reply:
There are two potential problems:

  - `invoke_on` racing with destroying an instance of the service
  - lambda passed to `invoke_on` racing with the destruction

I've fixed the second by using a gate on the destination core and created [an issue](https://github.com/vectorizedio/redpanda/issues/1861) to follow up the first problem (it's more fundamental and potentially affects all peering services)